crates/transpiler/src/passes/basis_translator/errors.rs

```diff
@@ -0,0 +1,85 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::PyErr;
+use qiskit_circuit::{circuit_data::CircuitError, error::DAGCircuitError};
+use thiserror::Error;
+
+use crate::TranspilerError;
+
+/// A collection of the Errrors possible in the [run_basis_translator] function
+#[derive(Debug, Error)]
+pub enum BasisTranslatorError {
+    #[error[
+        "Unable to translate the operations in the circuit: \
+        {basis} to the backend's (or manually specified) target \
+        basis: {expanded}. This likely means the target basis is not universal \
+        or there are additional equivalence rules needed in the EquivalenceLibrary being \
+        used. For more details on this error see: \
+        https://quantum.cloud.ibm.com/docs/api/qiskit/qiskit.transpiler.passes.\
+        BasisTranslator#translation-errors"
+    ]]
+    TargetMissingEquivalence { basis: String, expanded: String },
+    // TODO: Use rust native CircuitError
+    #[error[
+        "{0}"
+    ]]
+    BasisCircuitError(String),
+    // TODO: Use rust native GateError
+    #[error[
+    "{0}"
+    ]]
+    BasisGateError(String),
+    #[error[
+        "{0}"
+                ]]
+    // TODO: Use rust native DAGCircuitError
+    BasisDAGCircuitError(String),
+    #[error[
+        "BasisTranslator did not map {0}"
+    ]]
+    ApplyTranslationMappingError(String),
+    #[error[
+        "Translation num_params not equal to op num_params. \
+            Op: {:?} {} Translation: {:?}\n{:?}",
+            node_params,
+            node_name,
+            target_params,
+            target_dag
+    ]]
+    ReplaceNodeParamMismatch {
+        node_params: String,
+        node_name: String,
+        target_params: String,
+        target_dag: String,
+    },
+    #[error[
+        "Global phase must be real, but got {0}"
+    ]]
+    ReplaceNodeGlobalPhaseComplex(String),
+    #[error[
+        "Error mapping of parameters for replace_node: {0}"
+    ]]
+    ReplaceNodeParameterError(String),
```

Comment:
This one doesn't seem very intuitive to understand as user, could we just fail more generically here? Maybe grouped with the above and just say that parameter assignment during translation failed or sth?

Reply:
Yeah I can put this and the global phase error together. My only worry is that we lose specificity with the errors by generalizing so much. Ideally we'd want to use `this-error` to have a more holistic understanding of the error in question by having these variants. Since the Rust API is mostly developer facing, I think leaving these separate is okay.

That said, we can rename the ReplaceNodeParameterError to just re-use the `ParameterError` returned from the source and make it look more like this:
```rs
/// A collection of the Errrors possible in the [run_basis_translator] function
#[derive(Debug, Error)]
pub enum BasisTranslatorError {
    //...
    #[error(transparent)]
    BasisParameterError(#[from] ParameterError),   
    //...
}